public/js/SceneManager.js

```diff
@@ -52,9 +53,96 @@ class SceneManager {
     // current level layout and its tilesand tile
     this.currentLevel = LEVEL_ONE;
 
+    //music menu
+    this.musicMenu = new MusicMenu(this.gameEngine, 1050, 50, [
+      this.gameEngine.assetManager.getAsset('../img/music_menu.png'),
+      this.gameEngine.assetManager.getAsset('../img/music_menu_xmusic.png'),
+      this.gameEngine.assetManager.getAsset('../img/music_menu_xfx.png')
+    ]);
+
+  }
+
+  update() {
+    this.musicMenu.update();
   }
 
+  draw() {
+    this.musicMenu.draw();
+  }
+}
+
+/**
+* Class that holds a music menu to control the play and pause music backgrounds
+* and the sound fx. Also can pause and play music using the sound engine. 
+*/
+class MusicMenu {
+  constructor(game, xPosition, yPosition, images) {
+    this.gameEngine = game;
+    this.x = xPosition;
+    this.y = yPosition;
+    this.menu = images[0];
+    this.xMusic = images[1];
+    this.xFx = images[2];
 
+    this.ctx = this.gameEngine.ctx //not needed
+    this.xMusicChecked = false;
+    this.xFxChecked = false;
+
+  }
 
+  update() {
+    let clickXandY = this.gameEngine.click;
+    let changeInState = false;
+    if (clickXandY != null) {
+      // console.log(clickXandY);
+      //music x
+      if (
+        clickXandY.x >= this.x + 10 &&
+        clickXandY.x <= this.x + 70 &&
+        clickXandY.y >= this.y + 10 &&
+        clickXandY.y <= this.y + 20
+      ) {
+        changeInState = true;
+        this.xMusicChecked = this.xMusicChecked ? false : true;
+        // console.log("music checked", this.xMusicChecked);
+      }
+      //fx x
+      if (
+        clickXandY.x >= this.x + 10 &&
+        clickXandY.x <= this.x + 70 &&
+        clickXandY.y >= this.y + 30 &&
+        clickXandY.y <= this.y + 50
+      ) {
+        changeInState = true;
+        this.xFxChecked = this.xFxChecked ? false : true;
+        // console.log("fx checked", this.xFxChecked);
+      }
+    }
 
+    //play and pause pause audio if previous state has changed
+    if (changeInState) {
+      if (this.xMusicChecked) {
+        this.gameEngine.audio.pauseBackgroundMusic();
+      } else {
+        this.gameEngine.audio.unPauseBackgroundMusic();
+      }
+      //play or pause sound fx
+      if (this.xFxChecked) {
+        this.gameEngine.audio.muteSoundFX();
+      } else {
+        this.gameEngine.audio.unMuteSoundFX();
+      }
+    }
+  }
+
+  draw() {
+    //draw music music menu background
+    this.ctx.drawImage(this.menu, this.x, this.y);
+    if (this.xMusicChecked) {
+      this.ctx.drawImage(this.xMusic, this.x, this.y);
+    }
+    if (this.xFxChecked) {
+      this.ctx.drawImage(this.xFx, this.x, this.y);
```

Comment:
The clicking boxes are not quite lined up correctly. When I click in the upper left quadrant of the box, no change, when I click outside of the box to the lower right it 'checks' when it shouldn't.